### ISSUE TITLE
Support restore into higher Moodle version than the one in the backup

### ISSUE THREAD

**Issue by lmscloud-io** [lmscloud-io/moodle-tool_vault#8]
Currently when trying to restore a backup made in Moodle 4.1 in Moodle 4.3 , tool_vault fails a pre-check with a confusing error message, for example:

> Site version number has to be greater than or equal to the version in the backup. This backup is 2022112809.1 and this site is 2023100903.08

This is caused by this check https://github.com/lmscloud-io/moodle-tool_vault/blob/v1.4/classes/local/checks/version_restore.php#L57 that makes sure that the Moodle branch is equal AND the version is  greater or equal than the backup.

When branch comparison is removed, this pre-check reports a warning about mismatching version of every single standard plugin and the pre-check results are hard to read. 

Another problem would be the plugins that are removed, especially plugins with subplugins, such as `editor_tinymce`. All subplugins of the `tinymce` type will be left orphaned in the restored moodle, standard procedure to uninstall missing plugins in Moodle does not work when the plugin type is unknown. 

And finally an exception occurred during the upgrade

```
column "communication-&gt;contextid" cannot be modified. Dependency found with index "mdl_comm_con_ix (contextid)"

[More information about this error](https://docs.moodle.org/401/en/error/moodle/ddldependencyerror)

Debug info:
Error code: ddldependencyerror×Dismiss this notification
Stack trace:
line 734 of /lib/ddl/database_manager.php: ddl_dependency_exception thrown
line 591 of /lib/ddl/database_manager.php: call to database_manager->check_field_dependencies()
line 634 of /lib/ddl/database_manager.php: call to database_manager->change_field_type()
line 3672 of /lib/db/upgrade.php: call to database_manager->change_field_notnull()
line 1916 of /lib/upgradelib.php: call to xmldb_main_upgrade()
line 532 of /admin/index.php: call to upgrade_core()
```

Basically this issue is to make sure that restore is possible into other major version of moodle, the pre-check results are readable and no errors occur when restoring between any two Moodle versions that are supported by tool_vault
